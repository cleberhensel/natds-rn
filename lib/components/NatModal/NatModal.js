import React from 'react';
import { Modal } from 'react-native';
import PropTypes from 'prop-types';

import {
  ModalView,
  ModalContent,
  ModalText,
  ModalTitle,
  ModalButton,
  ModalButtonLabel,
  ModalContainer,
  ModalButtonAlternative,
  ModalButtonLabelAlternative,
} from './NatModal.styles';

export const NatModal = ({
  title,
  text,
  action,
  buttonText,
  alternative,
  buttonTextAlternative,
  alternativeAction,
  save,
}) => (
  <Modal
    animationType="fade"
    transparent
    onRequestClose={action}
    save={save}
  >
    <ModalView>
      <ModalContent accessibilityLabel={title}>
        <ModalTitle save={save}>{title}</ModalTitle>
        { !save && (<ModalText>{text}</ModalText>) }
        {
            alternative
              ? (
                <ModalContainer>
                  <ModalButtonAlternative
                    alternative={alternative}
                    onPress={alternativeAction}
                    accessibilityLabel={buttonTextAlternative.toUpperCase()}
                  >
                    <ModalButtonLabelAlternative
                      alternative={alternative}
                    >
                      {buttonTextAlternative.toUpperCase()}
                    </ModalButtonLabelAlternative>
                  </ModalButtonAlternative>

                  <ModalButton
                    alternative={alternative}
                    onPress={action}
                    accessibilityLabel={buttonText.toUpperCase()}
                  >
                    <ModalButtonLabel>{buttonText.toUpperCase()}</ModalButtonLabel>
                  </ModalButton>
                </ModalContainer>
              )
              : (
                <ModalButton
                  alternative={alternative}
                  onPress={action}
                  accessibilityLabel={buttonText.toUpperCase()}
                >
                  <ModalButtonLabel>{buttonText.toUpperCase()}</ModalButtonLabel>
                </ModalButton>
              )
          }
      </ModalContent>
    </ModalView>
  </Modal>
);

NatModal.propTypes = {
  action: PropTypes.func.isRequired,
  alternative: PropTypes.bool,
  alternativeAction: PropTypes.func,
  buttonText: PropTypes.string.isRequired,
  buttonTextAlternative: PropTypes.string,
  save: PropTypes.bool,
  text: PropTypes.string,
  title: PropTypes.string.isRequired,
};

NatModal.defaultProps = {
  alternative: false,
  alternativeAction: PropTypes.func,
  buttonTextAlternative: 'Button text error',
  save: false,
  text: 'Modal text error',
};
