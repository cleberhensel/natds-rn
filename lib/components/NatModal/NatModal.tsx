import React from "react";
import { Modal } from "react-native";
import {
  ModalView,
  ModalContent,
  ModalText,
  ModalTitle,
  ModalButton,
  ModalButtonLabel,
  ModalContainer,
  ModalButtonAlternative,
  ModalButtonLabelAlternative
} from "./NatModal.styles";
type NatModalProps = {
  action: (...args: any[]) => any,
  alternative?: boolean,
  alternativeAction?: (...args: any[]) => any,
  buttonText: string,
  buttonTextAlternative?: string,
  save?: boolean,
  text?: string,
  title: string
};
export const NatModal: React.SFC<NatModalProps> = ({
  title,
  text,
  action,
  buttonText,
  alternative,
  buttonTextAlternative,
  alternativeAction,
  save
}) => (
  <Modal animationType="fade" transparent onRequestClose={action} save={save}>
    <ModalView>
      <ModalContent accessibilityLabel={title}>
        <ModalTitle save={save}>{title}</ModalTitle>
        {!save && <ModalText>{text}</ModalText>}
        {alternative ? (
          <ModalContainer>
            <ModalButtonAlternative
              alternative={alternative}
              onPress={alternativeAction}
              accessibilityLabel={buttonTextAlternative.toUpperCase()}
            >
              <ModalButtonLabelAlternative alternative={alternative}>
                {buttonTextAlternative.toUpperCase()}
              </ModalButtonLabelAlternative>
            </ModalButtonAlternative>

            <ModalButton
              alternative={alternative}
              onPress={action}
              accessibilityLabel={buttonText.toUpperCase()}
            >
              <ModalButtonLabel>{buttonText.toUpperCase()}</ModalButtonLabel>
            </ModalButton>
          </ModalContainer>
        ) : (
          <ModalButton
            alternative={alternative}
            onPress={action}
            accessibilityLabel={buttonText.toUpperCase()}
          >
            <ModalButtonLabel>{buttonText.toUpperCase()}</ModalButtonLabel>
          </ModalButton>
        )}
      </ModalContent>
    </ModalView>
  </Modal>
);
NatModal.defaultProps = {
  alternative: false,
  alternativeAction: PropTypes.func,
  buttonTextAlternative: "Button text error",
  save: false,
  text: "Modal text error"
};
