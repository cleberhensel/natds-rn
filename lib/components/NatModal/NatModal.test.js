import React from 'react';
import renderer from 'react-test-renderer';

import { NatModal } from './NatModal';

describe('NatModal component and its variants', () => {
  describe('NatModal', () => {
    it('Should match snapshot', () => {
      const props = {
        action: jest.fn(),
        alternative: false,
        buttonText: 'Entendi',
        text: 'Texto de boas vindas',
        title: 'Boas vindas!',
        visible: true,
      };

      const modal = renderer.create(<NatModal {...props} />).toJSON();

      expect(modal).toMatchSnapshot();
    });
  });

  describe('NatModal error', () => {
    const props = {
      action: jest.fn(),
      alternative: true,
      alternativeAction: jest.fn(),
      buttonText: 'enviar link',
      buttonTextAlternative: 'cancelar',
      text: 'texto de descrição do modal',
      title: 'título',
    };

    it('should render the error mode', () => {
      const modalAlternative = renderer.create(<NatModal {...props} alternative />).toJSON();

      expect(modalAlternative).toMatchSnapshot();
    });
  });

  describe('NatModal save', () => {
    const props = {
      action: jest.fn(),
      buttonText: 'ok',
      save: true,
      text: 'não deve aparecer o text',
      title: 'título',
    };

    it('should render the save mode', () => {
      const modalAlternative = renderer.create(<NatModal {...props} save />).toJSON();

      expect(modalAlternative).toMatchSnapshot();
    });
  });
});
