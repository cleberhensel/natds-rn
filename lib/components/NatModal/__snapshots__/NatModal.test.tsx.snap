// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatModal component and its variants NatModal Should match snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  save={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(3, 3, 3, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityLabel="Boas vindas!"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6,
            "elevation": 8,
            "marginHorizontal": 24,
            "shadowColor": "rgba(0, 0, 0, 0.24)",
            "shadowOffset": Object {
              "height": 16,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 32,
          },
        ]
      }
    >
      <Text
        save={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "NaturaSans-Bold",
              "fontSize": 20,
              "lineHeight": 28,
              "marginBottom": 8,
              "marginTop": 24,
              "paddingHorizontal": 16,
            },
          ]
        }
      >
        Boas vindas!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 22,
              "marginBottom": 24,
              "paddingHorizontal": 16,
            },
          ]
        }
      >
        Texto de boas vindas
      </Text>
      <View
        accessibilityLabel="ENTENDI"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F4AB34",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "System",
                "letterSpacing": 1,
              },
            ]
          }
        >
          ENTENDI
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`NatModal component and its variants NatModal error should render the error mode 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  save={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(3, 3, 3, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityLabel="título"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6,
            "elevation": 8,
            "marginHorizontal": 24,
            "shadowColor": "rgba(0, 0, 0, 0.24)",
            "shadowOffset": Object {
              "height": 16,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 32,
          },
        ]
      }
    >
      <Text
        save={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "NaturaSans-Bold",
              "fontSize": 20,
              "lineHeight": 28,
              "marginBottom": 8,
              "marginTop": 24,
              "paddingHorizontal": 16,
            },
          ]
        }
      >
        título
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 22,
              "marginBottom": 24,
              "paddingHorizontal": 16,
            },
          ]
        }
      >
        texto de descrição do modal
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          accessibilityLabel="CANCELAR"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 6,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "50%",
            }
          }
        >
          <Text
            alternative={true}
            style={
              Array [
                Object {
                  "color": "#F4AB34",
                  "fontFamily": "System",
                  "letterSpacing": 1,
                },
              ]
            }
          >
            CANCELAR
          </Text>
        </View>
        <View
          accessibilityLabel="ENVIAR LINK"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F4AB34",
              "borderBottomLeftRadius": 0,
              "borderBottomRightRadius": 6,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "letterSpacing": 1,
                },
              ]
            }
          >
            ENVIAR LINK
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`NatModal component and its variants NatModal save should render the save mode 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  save={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(3, 3, 3, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityLabel="título"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6,
            "elevation": 8,
            "marginHorizontal": 24,
            "shadowColor": "rgba(0, 0, 0, 0.24)",
            "shadowOffset": Object {
              "height": 16,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 32,
          },
        ]
      }
    >
      <Text
        save={true}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "NaturaSans-Bold",
              "fontSize": 20,
              "lineHeight": 28,
              "marginBottom": 40,
              "marginTop": 40,
              "paddingHorizontal": 16,
            },
          ]
        }
      >
        título
      </Text>
      <View
        accessibilityLabel="OK"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F4AB34",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "System",
                "letterSpacing": 1,
              },
            ]
          }
        >
          OK
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
